### ISSUE TITLE
Favor request.delete() over request.del()

### ISSUE THREAD

**Issue by dennishall1** [visionmedia/superagent#605]
Request itself has to manage del as a special case of the REST verbs.  Most code that uses request also has to treat delete specially, as del.   Request could benefit from treating DELETE the same as all other verbs, and map it to request.delete instead of request.del.  (This update to request would align with how express treats delete:  http://expressjs.com/api.html#app.delete.method)

```
methods.forEach(function(method){
  var name = 'delete' == method ? 'del' : method;
```

**Comment by rase-** [comment]
I do think parity with `express` would make a lot of sense, not to mention consistency of verb method names, and would be in favor of this. 

cc @defunctzombie for an opinion

I wonder if there is another reason for missing `.delete` other than reserved names being generally considered bad practice for property names.

**Comment by defunctzombie** [comment]
I have been bitten by this a few times but the moment we change it, I am almost 100% certain someone will complain and say it doesn't work in `use strict` or something. As long as it works under `use strict` then I have no problem with making an alias.

**Comment by shesek** [comment]
It should be fine with strict mode as long as `obj['delete']` is used rather than `obj.delete`.

**Comment by dennishall1** [comment]
Shouldn't be anything wrong with `obj.delete`.  Strict mode should only disallow an attempt to set a value to a _global variable_ named `delete`, but this is not a global variable, it is an object's method name.

Seems fine:
`(function(){ "use strict";  var o = {};  o.delete = "hello"; return o; })();`
..."Use strict" does not appear to prevent setting an Object property named delete.

The only potential gotchya would be if the language ever decided to add an un-overridable `delete` method or property to the Object prototype.  `Object :: deleteProperty` is in the ES6 draft.

**Comment by defunctzombie** [comment]
Would review a PR for this.

**Comment by iMerica** [comment]
This has been closed, yet `.delete()` is still not supported. What was the final resolution?

**Comment by shesek** [comment]
@iMerica This issue is still opened, the "closed" you're seeing is from the referenced issue at [borisyankov/DefinitelyTyped](https://github.com/borisyankov/DefinitelyTyped).

**Comment by focusaurus** [comment]
Fixed by #743
